src/Scripting/Core/ScriptOptions.cs

```diff
@@ -282,5 +301,17 @@ private static MetadataReference CreateReferenceFromAssembly(Assembly assembly)
         /// <exception cref="ArgumentNullException"><paramref name="imports"/> is null or contains a null reference.</exception>
         public ScriptOptions AddImports(params string[] imports) =>
             AddImports((IEnumerable<string>)imports);
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with debugging information enabled.
+        /// </summary>
+        public ScriptOptions WithEmitDebugInformation(bool emitDebugInformation) =>
+            emitDebugInformation == EmitDebugInformation ? this : new ScriptOptions(this) { EmitDebugInformation = emitDebugInformation };
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with specified <see cref="FileEncoding"/>.
+        /// </summary>
+        public ScriptOptions WithFileEncoding(Encoding encoding) => 
```

Comment:
>WithFileEncoding [](start = 29, length = 16)

Just realized passing the encoding in options is not enough. The problem is that encodings are not 1:1 mappings in general. Say you have a script file encoded by encoding E saved on disk.  You read it to a string using decoder for E and then create a Script from that string and also pass in encoding E. Then to calculate the checksum for the source file that's stored in the PDB, we need to convert the string back to bytes using encoder for E. The problem is that for some encodings the resulting bytes might not be exactly the same that the file contains, unless the encoding is 1:1 mapping. 

Also, the compiler implements somewhat sophisticated encoding detection. So it would be unfortunate if the user needed to reimplement that. 

I'd suggest that we add overloads to core Scripting methods that take ```string code``` today. We don't need to add overloads to all methods since most of them are just convenience APIs that call the core ones.

I'm thinking these 3:

```C#
CSharpScript.Create(Stream code, ...)
Script.ContinueWith(Stream code, ...)
Script.ContinueWith<T>(Stream code, ...)
```

Then internally within Scripting API implementation we'd create SourceText using From(Stream, Encoding) method.

Could you give it a try?